crates/matrix-sdk-ui/src/timeline/controller/state.rs

```diff
@@ -624,7 +624,15 @@ impl TimelineStateTransaction<'_> {
                         }
 
                         RemoteEventOrigin::Pagination | RemoteEventOrigin::Cache => {
-                            // Forward the previous decision to add it.
+                            match self.timeline_focus {
+                                TimelineFocusKind::PinnedEvents => {
+                                    // Only insert timeline items for pinned events, if the event
+                                    // came from the sync.
+                                    should_add = room_data_provider.is_pinned_event(&event_id);
```

Comment:
This is a `match self.timeline_focus` in a `match origin`. Looking at the other branch of the `match origin`, it might be simpler to invert the two matches, and have the `match self.timeline_focus` be the outermost, while the `match origin` is the innermost? It would at least avoid the repetition for the pinned events focus.